saas/management/commands/report_weekly_revenue.py

```diff
@@ -49,172 +49,236 @@ def add_arguments(self, parser):
             dest='at_time', default=None,
             help='Specifies the time at which the command runs'
         )
-        parser.add_argument('--provider', action='append',
+        parser.add_argument(
+            '--provider', action='append',
             dest='providers', default=None,
-            help='Specifies provider to generate reports for.')
+            help='Specifies provider to generate reports for.'
+        )
+        parser.add_argument(
+            '--period', action='store',
+            dest='period', default='weekly',
+            choices=['hourly', 'daily', 'weekly', 'monthly', 'yearly'],
+            help='Specifies the period to generate reports for'
+        )
 
     @staticmethod
-    def construct_date_periods(at_time, timezone=None):
+    def construct_date_periods(at_time, period='weekly', timezone=None):
```

Comment:
1. Use [`humanize.WEEKLY`](https://github.com/djaodjin/djaodjin-saas/blob/3a2752cb0787ddf0562fc8ec7f091c7731174044/saas/humanize.py#L35) as default.

2. There are [month_periods](https://github.com/djaodjin/djaodjin-saas/blob/3a2752cb0787ddf0562fc8ec7f091c7731174044/saas/metrics/base.py#L48) and [day_periods](https://github.com/djaodjin/djaodjin-saas/blob/3a2752cb0787ddf0562fc8ec7f091c7731174044/saas/metrics/base.py#L111) already defined in saas/metrics/base.py. Maybe the whole construct method should be moved there?